### PR TITLE
docs: Prune old migration docs

### PR DESCRIPTION
**Pull request by aarnq**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [ ] personal data beyond what is necessary for interacting with this pull request, nor
> - [ ] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [x] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [x] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### Platform Administrator notice

Migration steps for old releases other than the last five releases have been pruned to simplify repository maintainability.
The old migrations can still be accessed by looking at their respective release tag.

### What does this PR do / why do we need this PR?

This might become a bit of discussion _if_ this is really something we even _should_ do, however I find it in our interest to prune old migration documentations.
Especially since it has been problematic from a maintainability perspective when enforcing linting and styling, as then the entire backlog of those needs to be updated each time a rule is more strict.
And since we've long lost context for how the platform looked like at that point, edits, even those caused by lint rules, might modify the content in some way that it potentially breaks.
So I want us to drop them, and adopt the "keep the last five releases rule".

Let me know what you all think.

#### Information to reviewers

This pruning will be part of a larger effort, as I intend to work on the change from migration/template to migration/main together with a script to promote main to the actual release that we had as an idea long ago, and this pruning will be automatic during that promotion.

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Documentation checks:
  - [ ] The [public documentation](https://github.com/elastisys/compliantkubernetes) required no updates
  - [ ] The [public documentation](https://github.com/elastisys/compliantkubernetes) required an update - [link to change](set-me\)
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Review by raviranjanelastisys — APPROVED** [review]

**Review by OlleLarsson — APPROVED** [review]
For me, removing the old migrations is OK.

File: migration/v0.38/README.md
Comment (viktor-f) [inline migration/v0.38/README.md:1]: Question/nit:
When you write "last 5 releases" I interpret that as the last 5 finished releases. That would currently include 0.38 since 0.43 is not released yet. Is that the intention or is the intention to have 5 releases including the last unreleased one?
I mostly just want to make sure I understand what you mean.

**Review by viktor-f — COMMENTED** [review]
I'm ok with this change.
But I would like some short text (perhaps in the release folder and/or migration folder) about this policy. What do you think about adding something for that? It could be very short and simple.

File: migration/v0.38/README.md
Comment (aarnq) [reply migration/v0.38/README.md:1]: Yes, I'll add to the readme about the versions kept and how to find old ones.

And yes, I probably should keep 0.38 now, as the intention is the five released ones.

Though the idea after this is merged is to rework the current migration WIP after 0.43 is released to become `main` and remove `template`. So I kind of want to finalise the structure without making the new method a dependency for the 0.43 release, as I assume we will start on it soon-ish.
Let me know if you think I should still keep 0.38 with that reasoning.

Comment (viktor-f) [reply migration/v0.38/README.md:1]: Ok, yes then it is ok to remove 0.38 (like it is now).

Will approve once the documentation is added :slightly_smiling_face:

**Review by lunkan93 — APPROVED** [review]

**Review by viktor-f — APPROVED** [review]

**Pull request merged**
